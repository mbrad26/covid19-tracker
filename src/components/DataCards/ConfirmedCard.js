import React from 'react';
import { Card } from 'react-bootstrap';

import ChartComponent from '../Chart/ChartComponent.js';

const ConfirmedCard = ({ cases, casesIncrease }) => (
  <Card className='shadow' id='cases'>
    <Card.Body>
      <Card.Title>Confirmed: <span className='text-warning font-weight-bold'>
          +{casesIncrease.toLocaleString()}
        </span>
      </Card.Title>
      <ChartComponent label={'Cases'} color={'orange'} />
      <Card.Text>
        {cases.toLocaleString()}
      </Card.Text>
    </Card.Body>
  </Card>
)

export default ConfirmedCard;
