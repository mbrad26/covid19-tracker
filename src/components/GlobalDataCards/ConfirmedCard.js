import React from 'react';
import { Card, Row, Col } from 'react-bootstrap';

import ChartComponent from '../../containers/ChartComponent';

const ConfirmedCard = ({ cases, casesIncrease }) => (
  <Card className='shadow' id='cases'>
    <Card.Body>
      <Card.Title>Confirmed: <span className='text-warning font-weight-bold'>
          +{casesIncrease.toLocaleString()}
        </span>
      </Card.Title>
      <ChartComponent />
      <Card.Text>
        {cases.toLocaleString()}
      </Card.Text>
    </Card.Body>
  </Card>
)

export default ConfirmedCard;
