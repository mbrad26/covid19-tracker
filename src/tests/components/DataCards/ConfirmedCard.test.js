import React from 'react';
// import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import { globalData, historicalData } from '../../fixtures';
import ConfirmedCard from '../../../components/DataCards/ConfirmedCard';

describe('ConfirmedCard', () => {
  it('renderes snapshot', () => {
    const { container } = render(
      <ConfirmedCard
        cases={globalData.cases}
        country={globalData.country ? globalData.country : 'Worldwide'}
        lastUpdated={new Date(globalData.updated).toDateString()}
        casesIncrease={globalData.todayCases}
        historicalData={historicalData}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
