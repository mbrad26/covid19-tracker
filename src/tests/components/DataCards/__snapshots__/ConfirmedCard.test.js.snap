// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmedCard renderes snapshot 1`] = `
<div
  class="card"
>
  <div
    class="card-body"
  >
    <div
      class="card-title h5"
    >
      Worldwide
      : 
      <span>
        <span>
          +0
        </span>
      </span>
      <i
        class="fas fa-frown-open"
      />
    </div>
    <canvas
      class="chartjs-render-monitor"
      height="0"
      style="display: block;"
      width="0"
    />
    <p
      class="card-text"
    >
      25,694,157
      <span
        class="text-muted"
        id="update"
      >
        Tue Sep 01 2020
      </span>
    </p>
  </div>
</div>
`;
